modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/CustomRXTSubscriptionTestCase.java

```diff
@@ -0,0 +1,285 @@
+        queryParamMap = new HashMap<String, String>();
+        headerMap = new HashMap<String, String>();
+        resourcePath =
+                FrameworkPathUtil.getSystemResourceLocation() + "artifacts" + File.separator + "GREG" + File.separator;
+        publisherUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "publisher/apis");
+        resourceAdminServiceClient = new ResourceAdminServiceClient(backendURL, session);
+        addCustomRxt();
+        new ServerConfigurationManager(automationContext).restartGracefully();
+        Thread.sleep(120000);
+        setTestEnvironment();
+    }
+
+    @Test(groups = { "wso2.greg",
+            "wso2.greg.es" }, description = "Adding subscription to custom asset on LC state change",
+            dependsOnMethods = { "addSubscriptionCheckListItem", "addSubscriptionUnCheckListItem" })
+    public void addSubscriptionToLcStateChange() throws JSONException, IOException {
+
+        JSONObject dataObject = new JSONObject();
+
+        dataObject.put("notificationType", "PublisherLifeCycleStateChanged");
+        dataObject.put("notificationMethod", "work");
+
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(publisherUrl + "/subscriptions/applications/" + assetId,
+                        MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON, dataObject.toString(), queryParamMap,
+                        headerMap, cookieHeader);
+
+        String payLoad = response.getEntity(String.class);
+        payLoad = payLoad.substring(payLoad.indexOf('{'));
+        JSONObject obj = new JSONObject(payLoad);
+        assertNotNull(obj.get("id").toString(),
+                "Response payload is not the in the correct format" + response.getEntity(String.class));
+
+        response = genericRestClient.geneticRestRequestPost(publisherUrl + "/assets/" + assetId + "/state",
+                MediaType.APPLICATION_FORM_URLENCODED, MediaType.APPLICATION_JSON,
+                "nextState=Testing&comment=Completed", queryParamMap, headerMap, cookieHeader);
+    }
+
+    @Test(groups = { "wso2.greg",
+            "wso2.greg.es" }, description = "Adding subscription to custom asset on resource update")
+    public void addSubscriptionToResourceUpdate() throws JSONException, IOException {
+
+        JSONObject dataObject = new JSONObject();
+
+        dataObject.put("notificationType", "PublisherResourceUpdated");
+        dataObject.put("notificationMethod", "work");
+
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(publisherUrl + "/subscriptions/applications/" + assetId,
+                        MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON, dataObject.toString(), queryParamMap,
+                        headerMap, cookieHeader);
+
+        String payLoad = response.getEntity(String.class);
+        payLoad = payLoad.substring(payLoad.indexOf('{'));
```

Comment:
Do we need this substring here. IMO we can create JSON object directly from payload